packages/material-ui/src/InputBase/InputBase.test.js

```diff
@@ -178,281 +210,230 @@ describe('<InputBase />', () => {
-        context: PropTypes.object,
-      };
-
-      wrapper = mount(<Provider />);
-    });
+    InputBaseWithContext.propTypes = {
+      context: PropTypes.object,
+    };
 
     it('should have the formControl class', () => {
-      setFormControlContext({});
-      assert.strictEqual(findOutermostIntrinsic(wrapper).hasClass(classes.formControl), true);
+      const { container } = render(<InputBaseWithContext context={{}} />);
+      expect(container.firstChild).to.have.class(classes.formControl);
     });
 
     describe('callbacks', () => {
-      let focus;
-
-      beforeEach(() => {
-        focus = spy();
-        wrapper.find('input').instance().focus = focus;
-        setFormControlContext({
-          onFilled: spy(),
-          onEmpty: spy(),
-          onFocus: spy(),
-          onBlur: spy(),
-        });
-      });
-
       it('should fire the onFilled muiFormControl and props callback when dirtied', () => {
         const handleFilled = spy();
-        wrapper.setProps({
-          onFilled: handleFilled,
-        });
-
-        wrapper.find('input').instance().value = 'hello';
-        wrapper.find('input').simulate('change', { target: { value: 'hello' } });
-        assert.strictEqual(handleFilled.callCount, 1);
-        assert.strictEqual(muiFormControl.onFilled.callCount, 1);
+        const muiFormControl = { onFilled: spy() };
+        const { container } = render(
+          <InputBaseWithContext context={muiFormControl} onFilled={handleFilled} />,
+        );
+
+        fireEvent.change(container.querySelector('input'), { target: { value: 'hello' } });
+        expect(handleFilled.callCount).to.equal(1);
+        expect(muiFormControl.onFilled.callCount).to.equal(1);
       });
 
       it('should fire the onEmpty muiFormControl and props callback when cleaned', () => {
         const handleEmpty = spy();
+        const muiFormControl = { onEmpty: spy() };
+        const { container } = render(
+          <InputBaseWithContext context={muiFormControl} onEmpty={handleEmpty} />,
+        );
 
         // Set value to be cleared
-        wrapper.find('input').instance().value = 'test';
-        wrapper.find('input').simulate('change', {
-          target: {
-            value: 'test',
-          },
-        });
-        wrapper.setProps({
-          onEmpty: handleEmpty,
-        });
-        assert.strictEqual(handleEmpty.callCount, 0);
+        fireEvent.change(container.querySelector('input'), { target: { value: 'test' } });
+        expect(handleEmpty.callCount, 0);
+
         // Clear value
-        wrapper.find('input').instance().value = '';
-        wrapper.find('input').simulate('change', {
-          target: { value: '' },
-        });
-        assert.strictEqual(handleEmpty.callCount, 1);
-        assert.strictEqual(muiFormControl.onEmpty.callCount, 2);
+        fireEvent.change(container.querySelector('input'), { target: { value: '' } });
+        expect(handleEmpty.callCount).to.equal(2);
```

Comment:
Was 1 is now 2 because of implicit act. Those values don't really make sense to me right now but I will follow-up on this because the current behavior probably causes #16458.